python/semantic_kernel/connectors/ai/onnx/onnx_utils.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) Microsoft. All rights reserved.
+from enum import Enum
+
+from semantic_kernel.contents import AuthorRole, ChatHistory, ImageContent, TextContent
+from semantic_kernel.exceptions import ServiceException, ServiceInvalidRequestError
+
+
+class ONNXTemplate(Enum):
+    """ONNXTemplate is an enumeration that represents different ONNX model templates.
+
+    Attributes:
+        PHI3 (str): Represents the "phi3" ONNX model template.
+        PHI3V (str): Represents the "phi3v" ONNX model template.
+        GEMMA (str): Represents the "gemma" ONNX model template.
+        LLAMA (str): Represents the "llama" ONNX model template.
+    """
+
+    PHI3 = "phi3"
+    PHI3V = "phi3v"
+    GEMMA = "gemma"
+    LLAMA = "llama"
+
+
+def apply_template(history: ChatHistory, template: ONNXTemplate) -> str:
+    """Apply the specified ONNX template to the given chat history.
+
+    Args:
+        history (ChatHistory): The chat history to which the template will be applied.
+        template (ONNXTemplate): The ONNX template to apply. Possible values are:
+            - ONNXTemplate.PHI3
+            - ONNXTemplate.GEMMA
+            - ONNXTemplate.LLAMA
+            - ONNXTemplate.PHI3V
+    Returns:
+        str: The result of applying the template to the chat history.
+
+    Raises:
+        ServiceException: If an error occurs while applying the template.
+    """
+    try:
+        if template == ONNXTemplate.PHI3:
```

Comment:
this will be cleaner if replaced with a dict with template as key, func as value, which is then invoked.